### DISCUSSION TITLE
Agama UX: More clickable items

### DISCUSSION THREAD

**Discussion by shundhammer** [agama-project/agama#1784] [Ideas]
## Summary

Please make more items on the screen clickable, even if it's only as an additional shortcut to navigate to another page. This would not only make the operation smoother, it would also avoid mental acrobatics to figure out how and where to change a setting that the user wants to change.

## Existing example

![agama-main-01-overview-initial](https://github.com/user-attachments/assets/a92082ed-991d-4307-bc44-c03ed906d58b)

In the "Installation blocking issues" box, "Defining a user..." is already clickable: it leads to the "Users" page.

While strictly this is redundant since the "Users" page is directly accessible from this page (it's only one click away), it's a good thing to have this here as well, since the user's mind is concentrated on those "blocking issues"; the user will want to get those things fixed ASAP.

## Other places

![agama-main-02-overview](https://github.com/user-attachments/assets/4d342bde-0d03-4703-ac32-8408c9dbbcc8)

- The system language also looks very much like a button, so a user might think that it's clickable (for sure I thought it was, and I tried clicking), but no, it's not: It's purely a passive element.

  Oddly enough, the keyboard layout is not listed here; many users will want to stick with English, but their native keyboard layout is a must-have; but this cannot easily be verified here.

- The storage device might also be a candidate if there are more than one to choose from.

- Software: Of course the installation size would not make any sense to click, but the additional software patterns might.

![agama-storage-01-initial](https://github.com/user-attachments/assets/bb32b389-7d5a-412a-8e26-4761fb1de20a)

- The planned partitions in the "Partitions and file systems" box just _cry_ "click me" to the user. That would be the natural interaction here: When I wanted to change the root filesystem type away from Btrfs (see issue #1758), this was the first thing that I tried.

  I am not sure what I'd expect to happen here; but it should do _something_. Seeing the object of my desires in front of my nose and refusing to do anything when I click on it is just... frustrating, to say the least.

- "There are 4 destructive actions planned" - this is alread highlighted, so it also cries "click me!". But nothing happens when I click on this. Importantly, of course this should _only_ show the destructive actions, not all 32 ones that include a ton of subvolume actions (which I am really almost never interested in).

- The table in the "final layout" box. This looks very much like what we always had in YaST, and it was an interactive table / tree in YaST where the user could fine-tune pretty much everything about partitioning.

  Knowing the structure and idea behind this, I can understand that this is now a purely passive thing. But again, this is frustrating, as I already have the thing that I might want to change directly in front of me.

  If nothing else, can we please at least add a "Change" button or link here that leads... somewhere? (To be discussed)

## Navigation - far away and back again

Having a link that navigates to another place, possibly to another page, is the trivial thing. Not so trivial is to make sure that the user doesn't completely use the context: Where exactly was I before that link that I clicked so casually took me away to another dimension? How do I get back?

Right now, Agama's pages aren't so complex that this would be a major problem. We hope that it stays that way, but realistically, more and more things will be added over time, making the "site map" more complex.

But actually, we already have this problem: The first page mentioned here with the "blocking issues" already causes a hyperjump to a completely different location. 

### Back and forward buttons

Users who are aware that they are actually using an application running inside an Internet browser (Firefox in this case) might get the idea to use their back / forward mouse buttons (if the mouse has any), or the common `Alt` + `Cursor Left` and `Alt` + `Cursor Right` keyboard shortcuts.

But that's the kind of users who (hopefully) also don't get lost so easily to begin with. Or the kind of users who are blissfully unaware that there is a difference between browsing the web and any other application, and just use those mouse buttons or keyboard shortcuts regardless.

**Comment by dgdavid** [comment]
> ## Summary
> Please make more items on the screen clickable, even if it's only as an additional shortcut to navigate to another page. This would not only make the operation smoother, it would also avoid mental acrobatics to figure out how and where to change a setting that the user wants to change.

Although I can agree at some extent, we have to be careful because this is not always true. Crowding the UI of links might not be a good idea for all users, specially keyboard users.


> 
> ## Existing example
> [screenshot]
> 
> In the "Installation blocking issues" box, "Defining a user..." is already clickable: it leads to the "Users" page.
> 
> While strictly this is redundant since the "Users" page is directly accessible from this page (it's only one click away), it's a good thing to have this here as well, since the user's mind is concentrated on those "blocking issues"; the user will want to get those things fixed ASAP.

Sure. Here the link make sense. BTW, these warnings will be moved to a more accessible place, already commented in a Trello card, see https://trello.com/c/tK4j4YUY/617-agama-show-installation-issues-from-the-top-bar (internal link)

> 
> ## Other places
> [screenshot]
> 
> * The system language also looks very much like a button, so a user might think that it's clickable (for sure I thought it was, and I tried clicking), but no, it's not: It's purely a passive element.

Here we do not need a link for such a label. No, please. What is more, this is gonna change too, not sure if the Trello card is already created. Basically, there will be links, but in the section name. Which could be quite useful when the sidebar is hidden. Users at overview will not need to expand the sidebar for navigating. Will elaborate the idea when hace time. But remember that this Overview page was made wit the promise to be rethink when possible, since we were shaping other parts and had  a lot of doubts about this page at the same time that needed something.


>   Oddly enough, the keyboard layout is not listed here; many users will want to stick with English, but their native keyboard layout is a must-have; but this cannot easily be verified here.

As said, such a page have to be revamped. However, we should think deeply if we really want to have everything in the overview... Because then we will be (again) in navigation-less application.

> * The storage device might also be a candidate if there are more than one to choose from.
> * Software: Of course the installation size would not make any sense to click, but the additional software patterns might.
> 

Same than above. The overview has been kind of _temporary_ since we decided to change the Agama UI. Of course, your comments will help, but we should invest time in order to have something more stable.


(Skipping the rest of the answer by now, will continue tomorrow. BTW, please remember that we are already working in a storage revamp)

**Comment by dgdavid** [comment]
> not sure if the Trello card is already created

Yes, it was there, although _unfinished_ (lack of time, you know). See it at https://trello.com/c/Cm6AUD2Y/618-agama-re-do-the-overview-page (internal link)

**Comment by dgdavid** [comment]
> * The planned partitions in the "Partitions and file systems" box just _cry_ "click me" to the user. That would be the natural interaction here: When I wanted to change the root filesystem type away from Btrfs (see issue [Agama UX: It's very hard to switch away from Btrfs #1758](https://github.com/agama-project/agama/issues/1758)), this was the first thing that I tried.
>   I am not sure what I'd expect to happen here; but it should do _something_. Seeing the object of my desires in front of my nose and refusing to do anything when I click on it is just... frustrating, to say the least.


Thanks for the observation, but as you know we already realized this was complex and it gonna be completely changed. In fact, we've been shaping the new approach for more than a month already, since it has a lot of changes in the backend. So, hope you don't mind but this is not relevant anymore for me. Don't get me wrong, knowing what does not work is good, but now I prefer focusing in the new approach.


> * "There are 4 destructive actions planned" - this is alread highlighted, so it also cries "click me!". But nothing happens when I click on this.

Again, known bug. There is already an issue open about it :cry:, properly labelled as `bug`. I added no comment there for the reason given above: I'm trying to focus in the new approach, which will solved the problem.

> Importantly, of course this should _only_ show the destructive actions, not all 32 ones that include a ton of subvolume actions (which I am really almost never interested in).

I really sorry, but this is a preference. And it is good and welcome, but we cannot transform our preferences in an UI, because every user has preferences and I fear we couldn't satisfy all of them :wink: Of course, we will try to come with an UI which will _emphasize_ the objectively important things. But I see no reason for not providing to the user a way to inspect these subvolume actions. I'm really confident we can provide them there without disturbing users not interested on them.

Just as a note: I'm trying to avoid the `I whatever` when `I` is used as `I as user` because IMHO we don't know which is important for who. In your example, "I am really almost never interested in". I got your point, of course. And I actually agree with you. I'm not interested in those actions neither. But we have room for placing them there. Bad example, but what I mean is that we should try to think the UI without overdoing assumptions.


> * The table in the "final layout" box. This looks very much like what we always had in YaST, and it was an interactive table / tree in YaST where the user could fine-tune pretty much everything about partitioning.
>   Knowing the structure and idea behind this, I can understand that this is now a purely passive thing. But again, this is frustrating, as I already have the thing that I might want to change directly in front of me.

This lead to the expert partitioner, which is something Agama will not have at least in the mid-term as far as I know. I could be aligned with your complain if we were trying to provide something similar to the YaST partitioner, but it is not the case by now. Not my decision :wink: 

>   If nothing else, can we please at least add a "Change" button or link here that leads... somewhere? (To be discussed)

We can add as many hints as needed to reduce the cognitive overload of such a section, of course. Not sure if mentioned button is an option because there is not an specific somewhere at this moment. Maybe an explanatory text could be a good starting point. The things is that we are not there yet. WIP.

**Comment by dgdavid** [comment]
> ## Navigation - far away and back again
> Having a link that navigates to another place, possibly to another page, is the trivial thing. Not so trivial is to make sure that the user doesn't completely use the context: Where exactly was I before that link that I clicked so casually took me away to another dimension? How do I get back?
> 

Not sure if I got it completely.

Hard to me to think in users as lost as you describe. I can be wrong, but in its current shape Agama offer users a complete freedom to go whenever they want, without restrictions. Agama UI it's not a wizard. Users are not required to follow specific steps. Even when user are editing an already set setting, in a page with these "Cancel" and "Accept" actions, users can regret of what they are doing and jump to completely different section. 

> Right now, Agama's pages aren't so complex that this would be a major problem. We hope that it stays that way, but realistically, more and more things will be added over time, making the "site map" more complex.

Honestly, I prefer waiting. If we start assuming we are not able to shape a simple UI for complex actions... we will deliver a complex UI for simple actions.

> But actually, we already have this problem: The first page mentioned here with the "blocking issues" already causes a hyperjump to a completely different location.

I disagree. We do not have a problem there. But, as commented, there are already plans for improving such a "hyperjump", see https://trello.com/c/Cm6AUD2Y/618-agama-re-do-the-overview-page (internal link)

**Comment by dgdavid** [comment]
> ### Back and forward buttons
> Users who are aware that they are actually using an application running inside an Internet browser (Firefox in this case) might get the idea to use their back / forward mouse buttons (if the mouse has any), or the common `Alt` + `Cursor Left` and `Alt` + `Cursor Right` keyboard shortcuts.
> 
> But that's the kind of users who (hopefully) also don't get lost so easily to begin with. Or the kind of users who are blissfully unaware that there is a difference between browsing the web and any other application, and just use those mouse buttons or keyboard shortcuts regardless.

Let me insist: Agama UI is not a wizard :wink:

**Comment by dgdavid** [comment]
@shundhammer 

First of all, thanks a ton for your input. and the time you have invested on it. I have to admit I would have liked to have such a valuable feedback many months ago :sweat_smile:, since it could have helped us avoid certain things.

That said, I hope you take my answers in a kind tone. I'm not opposed to making changes in the proposed directions, but first I'd like to think (a lot) about it before continuing to write UI code in a hurry. In spite of the fact that working in such a way allows us to deliver features and get feedback faster, I strongly believe we can slow down to make the cooking better :) But this is just my opinion, not a team decision.

Last but not least, excuse me for giving a partial answer yesterday afternoon :innocent:

See you around. Keep going!

**Comment by dgdavid** [comment]
The Agama web interface has been evolving and has recently gone through some changes :wink:. Given the improvements in navigation, I believe we can close this discussion for now. It's likely that further enhancements will continue to be made as well. 

Thanks!
